executor/src/witgen/data_structures/finalizable_data.rs

```diff
@@ -450,39 +377,62 @@ impl<T: FieldElement> FinalizableData<T> {
             },
         )
     }
+
+    pub fn get_in_progress_row(&self, i: usize, make_fresh_row: impl Fn() -> Row<T>) -> Row<T> {
```

Comment:
Would it be better to return something else in the finalized case, that is then processed by a function in the caller? (looking at the weird `make_fresh_row` function)

Reply:
I made some changes (e.g. `FinalizableData` having access to `FixedData`) so that it create the row itself.